guarddog/scanners/npm_package_scanner.py

```diff
@@ -18,35 +18,54 @@ def __init__(self) -> None:
         super().__init__(Analyzer(ECOSYSTEM.NPM))
 
     def download_and_get_package_info(self, directory: str, package_name: str, version=None) -> typing.Tuple[dict, str]:
-        git_target = None
-        if urlparse(package_name).hostname is not None and package_name.endswith('.git'):
-            git_target = package_name
+        git_package_name = self._parse_git_url(package_name)
 
-        if not package_name.startswith("@") and package_name.count("/") == 1:
-            git_target = f"https://github.com/{package_name}.git"
+        if git_package_name != "":
+            data, tarball_url = self._get_git_data_and_tarball_url(git_package_name, version)
+        else:
+            data, tarball_url = self._get_npm_data_and_tarball_url(package_name, version)
 
-        if git_target is not None:
-            raise Exception("Git targets are not yet supported for npm")
+        log.debug(f"Downloading NPM package from {tarball_url}")
+        file_extension = pathlib.Path(tarball_url).suffix
+        zippath = os.path.join(directory, package_name.replace("/", "-") + file_extension)
+        unzippedpath = zippath.removesuffix(file_extension)
+        self.download_compressed(tarball_url, zippath, unzippedpath)
+
+        return data, unzippedpath
+
+    def _parse_git_url(self, package_name: str) -> str:
```

Comment:
nit: would be good to have an unit test for that one (not a blocker though)

Reply:
Good point, updated one existing (but skipped) test in [b20a248](https://github.com/DataDog/guarddog/pull/526/commits/b20a248863206cae29c426a692ba90741ebfb9d9).